### PR TITLE
Fixes test which fails for some timezone or locale

### PR DESCRIPTION
**Pull request by damianszczepanik**

### CODE REVIEW

**Comment by damianszczepanik** [comment]
Should unblock releasing process

File: src/test/java/jenkins/plugins/office365connector/TimeUtilsTest.java [hunk 1]
```diff
@@ -13,15 +13,19 @@
...
         // when
         String format = TimeUtils.dateToString(date);
 
         // then
         // validates only the date as the timezone and local can be different
-        assertThat(format).startsWith("2017-09-29");
-        // because of the timezone, hours depends whete the test is executed
+        assertThat(format).contains("2017");
+        assertThat(format).contains("09");
+        assertThat(format).contains("29");
```
Comment (casz) [inline src/test/java/jenkins/plugins/office365connector/TimeUtilsTest.java]: Don't see it fixing the test. When expected output was "2017-09-30 02:05:41 IST" 29 is not contained with that expected output :)

Comment (damianszczepanik) [reply src/test/java/jenkins/plugins/office365connector/TimeUtilsTest.java]: Not sure what do you mean. I selected date so it makes sense and is not affected by the locals or timezone. Read assumptions above.

Comment (casz) [reply src/test/java/jenkins/plugins/office365connector/TimeUtilsTest.java]: Test is easily breakable.
Switching timezone from CEST works, to IST (Indian standard time)
Timezone can switch days too not only hours 👍 
```
java.lang.AssertionError: 
Expecting:
 <"2017-09-30 02:05:41 IST">
to contain:
 <"29"> 

	at jenkins.plugins.office365connector.TimeUtilsTest.dateToString_ReturnsDateAsString(TimeUtilsTest.java:27)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)```

Comment (casz) [reply src/test/java/jenkins/plugins/office365connector/TimeUtilsTest.java]: I would use a setup and teardown for this unit test.
In setup set default to UTC and in teardown set it back to default 👍

Comment (casz) [reply src/test/java/jenkins/plugins/office365connector/TimeUtilsTest.java]:
```java
    static TimeZone defaultTimeZone;

    @BeforeClass
    public static void setup() {
        defaultTimeZone = TimeZone.getDefault();
        TimeZone.setDefault(TimeZone.getTimeZone("UTC"));
    }

    @AfterClass
    public static void tearDown() {
        TimeZone.setDefault(defaultTimeZone);
    }
```

Comment (casz) [reply src/test/java/jenkins/plugins/office365connector/TimeUtilsTest.java]: Documentation might help:

No reason for teardown I guess 👍 
```
/**
     * Sets the {@code TimeZone} that is returned by the {@code getDefault}
     * method. {@code zone} is cached. If {@code zone} is null, the cached
     * default {@code TimeZone} is cleared. This method doesn't change the value
     * of the {@code user.timezone} property.
     *
     * @param zone the new default {@code TimeZone}, or null
     * @throws SecurityException if the security manager's {@code checkPermission}
     *                           denies {@code PropertyPermission("user.timezone",
     *                           "write")}
     * @see #getDefault
     * @see PropertyPermission
     */
```

Comment (SrhebbarMicrosoft) [inline src/test/java/jenkins/plugins/office365connector/TimeUtilsTest.java]: Am not sure if this fixes the issue. Still the expected and the actual output is different.

Comment (damianszczepanik) [reply src/test/java/jenkins/plugins/office365connector/TimeUtilsTest.java]: Removed one more assertion as it strongly depends of the locale or timezone. On the other hand I don't want to set timezone or mock it as the output of this method strongly depends just from that variable. So mocking is making useless this test IMO

Comment (SrhebbarMicrosoft) [reply src/test/java/jenkins/plugins/office365connector/TimeUtilsTest.java]: You need to mock the locale and test for different locale the format is correct. Just commenting the day is not enough. If the test is executed on end of month, then even month changes. If executed at the end of the year even the year changes. 
As even doing that would be just like testing the library function, I don't see the point of this test case itself.

Comment (damianszczepanik) [reply src/test/java/jenkins/plugins/office365connector/TimeUtilsTest.java]: Good point, current approach is very naive. Will apply suggestion provided by @casz then it should be enough to validate all elements or even whole string at once.

**Comment by damianszczepanik** [comment]
Should work fine now. Added comments with explanation.

**Pull request merged**